File: src/fensu/config/main/configuration_directory.py
"""Locate the configuration owner from a validated target-local project root."""

from pathlib import Path


def configuration_directory(*, project_root: Path, target_root: str) -> Path:
    """Ascend the normalized relative target path without discovering another repository."""

    directory: Path = project_root
    for _ in Path(target_root).parts:
        directory = directory.parent
    return directory

File: crates/fensu-cli/src/configuration/main/configuration_directory.rs
//! Locate the configuration owner of a target-local project root.

use std::path::{Component, Path, PathBuf};

pub(crate) fn configuration_directory(project_root: &Path, target_root: &str) -> PathBuf {
    let mut directory = project_root.to_path_buf();
    for component in Path::new(target_root).components() {
        if matches!(component, Component::Normal(_)) {
            directory.pop();
        }
    }
    directory
}
